WordSearch.java

```diff
@@ -0,0 +1,113 @@
+import java.util.*;
+
+/**
+ * Created by Alice Colt.
+ */
+public class WordSearch {
+
+
+    // Search for a WORD - using list
+
+    static List<String> wordSearchGrid(char[][] characters,
+                                       int row,
+                                       int column,
+                                       Dictionary dictionary){
+
+
+        // List for keeping all the solutions
+        List<String> solution = new LinkedList<>();
+
+        boolean[][] visited = new boolean[row][column];
+
+
+        for (int index = 0; index < row; index++)
+        {
+            for (int indexj = 0; indexj < column; indexj++)
+            {
+                Set<String> validWord = wordGetter(Character.toString(characters[index][indexj]),
+                                                  index, indexj,
+                                                  row, column,
+                                                  characters,visited,dictionary);
+
+                // Add to solution list
+                solution.addAll(validWord);
+            }//for
+        }//for
+
+
+        // Return solution list
+        return new ArrayList<>(solution);
+
+    }// List wordSearchInGrid
+
+
+
+    // Get word - using hashes
+
+    static Set<String> wordGetter(String prefix,
+                                  int indexRow, int row,
+                                  int indexColumn, int column,
+                                  char[][] characters,
+                                  boolean[][] visited,
+                                  Dictionary dictionary)
+    {
+
+        // Set for keeping all solutions
+        Set<String> solution = new HashSet<>();
+
+        // If prefix is part of a dictionary word, add to solutions.
+        if(dictionary.isWord((prefix)))
```

Comment:
Do you mean isPrefix or isWord here? You want to add it to the solution if it's a word, but only want to return if it's not a prefix.